packages/itwin/tree-widget/src/tree-widget-react/components/trees/common/internal/UseActiveViewport.tsx

```diff
@@ -4,21 +4,33 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { useEffect, useState } from "react";
-import type { ScreenViewport } from "@itwin/core-frontend";
 import { IModelApp } from "@itwin/core-frontend";
+import { createTreeWidgetViewport } from "../TreeWidgetViewport.js";
+
+import type { ScreenViewport } from "@itwin/core-frontend";
+import type { TreeWidgetViewport } from "../TreeWidgetViewport.js";
 
 /**
  * A copy of `useActiveViewport` from `@itwin/appui-react` package until X is fixed.
  * @internal
  */
-export function useActiveViewport(): ScreenViewport | undefined {
-  const [activeViewport, setActiveViewport] = useState(IModelApp.viewManager.selectedView);
+export function useActiveViewport({ treeWidgetViewport }: { treeWidgetViewport?: TreeWidgetViewport }): TreeWidgetViewport | undefined {
+  const [activeViewport, setActiveViewport] = useState(treeWidgetViewport ?? createTreeWidgetViewportInternal(IModelApp.viewManager.selectedView));
   useEffect(() => {
-    setActiveViewport(IModelApp.viewManager.selectedView);
-    return IModelApp.viewManager.onSelectedViewportChanged.addListener((args) => {
-      setActiveViewport(args.current);
-    });
-  }, []);
+    if (treeWidgetViewport) {
+      setActiveViewport(treeWidgetViewport);
+      return;
+    } else {
+      setActiveViewport(createTreeWidgetViewportInternal(IModelApp.viewManager.selectedView));
+      return IModelApp.viewManager.onSelectedViewportChanged.addListener((args) => {
+        setActiveViewport(createTreeWidgetViewportInternal(args.current));
+      });
+    }
```

Comment:
don't need an `else` if there's a `return`


```suggestion
    }
    setActiveViewport(createTreeWidgetViewportInternal(IModelApp.viewManager.selectedView));
    return IModelApp.viewManager.onSelectedViewportChanged.addListener((args) => {
      setActiveViewport(createTreeWidgetViewportInternal(args.current));
    });
```